Sources/SwiftFormat/Core/RuleMask.swift

```diff
@@ -111,10 +111,11 @@ enum IgnoreDirective: CustomStringConvertible {
     }
   }
 
-  /// Regex pattern to match an ignore comment. This pattern supports 0 or more comma delimited rule
-  /// names. The rule name(s), when present, are in capture group #3.
-  fileprivate var pattern: String {
-    return #"^\s*\/\/\s*"# + description + #"((:\s+(([A-z0-9]+[,\s]*)+))?$|\s+$)"#
+  /// Regex pattern to match an ignore directive comment.
+  /// - Capture group #1 captures the rule names if `":"` is present.
+  fileprivate func makeRegex() -> Regex<(Substring, Substring?)> {
+    let pattern = #"^\s*\/\/\s*"# + description + #"(?:\s*:\s*(.+))?$"#
```

Comment:
If we can't use RegexBuilder here because of platform limitations, it would still be nice to use real regex literals.

I don't remember from the previous PR, but if the `description` property is only used here, you could lift the regexes up into that instead, something like:

```swift
var regex: Regex {
  switch self {
  case .node: return /...swift-format-ignore.../
  case .file: return /...swift-format-ignore-file.../
}
```

Reply:
It would also be great to use named capture groups so that the result tuples elsewhere get labels, to make it more readable.

Reply:
`description` is used in `OrderedImports` to distinguish directives. 
Also, after modifying it to use regex literals and running the CI, I found that, as @ahoppen mentioned below, regex literals are not supported on Windows Swift 5.9 (5.10). Given this, it seems best to keep the current implementation as is.

Reply:
> It would also be great to use named capture groups so that the result tuples elsewhere get labels, to make it more readable.

I have utilized the named capture group in the regex to add the label `ruleNames`.

Reply:
Every time we try to do some good, Windows is there lurking in the background. 🙁 

Thanks for checking!